### ISSUE TITLE
ckains holds unnecessary mountpoints which appear busy from the parent

### ISSUE THREAD

**Issue by guillon** [mickael-guene/ckains#3]
The implementation does not correctly release mount points in the parent rootfs which then remain in the created namespace and appear busy from the parent point of view. Even if they are not bound into the new namespace.

This cause issues when the actual host need to unmount some FSs, in particular this causes troubles to the autofs implementation that is not able to correctly expire/unmount some autofs points while actually the running containers do not need them. 

For the implementation of bindings, the current implementation (as of v1.0.2) creates a temporary root which binds the parent rootfs to an old rootfs bind point, chroot to the new rootfs and mounts bindings from the old rootfs bind.
There is an issue which this implementation as actually the old rootfs bind point holds the parent rootfs for the duration of the ckains execution.
I.e. actually the unmounting (with MNT_DETACH) as no effect at all.

In order to let the new rootfs free the parent mounts, one must move the parent rootfs to the created old rootfs point, as would do a mount MS_MOVE or a pivot_root() syscall.

Reproduce the problem by exhibiting for instance some dangling mount in the ckains process with:

```
# Get into a directory which is a tmpfs (to be sure that we don't need to hold more than '/')
host$ cd /dev/shm
# copy ckains there
host$ cp .../ckains .
# get a busybox image and extract it
host$ wget -O rootfs.tar https://github.com/jpetazzo/docker-busybox/raw/master/rootfs.tar
host$ mkdir rootfs
host$ tar xvf rootfs.tar -C rootfs || true
# Execute ckains with only the /proc binding (to see mounts at least)
host$ ./ckains -r rootfs -b /proc -w /
# In the new NS we see only /proc and sub-mounts of /proc in mountinfo
$ cat /proc/self/mountinfo
... /proc
# BUT, when looking at actual mounts, we basically see all parent mounts
# For instance sysfs which was no bound
$ cat /proc/self/mounts | grep sysfs
sysfs /sys sysfs rw,relatime 0 0
# I.E., actually 37 mount points on my host
$ cat /proc/self/mounts | wc -l
37
```

We should expect to see only the mount points under /proc and the original rootfs root '/' itself in /self/proc/mounts, not the 37 mounts points here. They are actually all inherited from the parent and could not be unmounted properly, hence they are now all busy from the parent point of view.

**Comment by guillon** [comment]
Proposed fix in pull request #4.

The proposed implementation uses the pivot_root() syscall.

With respect to previous comment and this version of ckains, here is the output of /proc/self/mounts:

```
host$ cd /dev/shm
host$ ./ckains -r rootfs -b /proc -w /
$ cat /proc/self/mounts | wc -l
4
$ cat /proc/self/mounts | wc -l
rootfs / rootfs rw 0 0
tmpfs / tmpfs rw,nosuid,nodev 0 0
proc /proc proc rw,relatime 0 0
systemd-1 /proc/sys/fs/binfmt_misc autofs rw,relatime,fd=34,pgrp=1,timeout=300,minproto=5,maxproto=5,direct 0 0
nfsd /proc/fs/nfsd nfsd rw,relatime 0 0
```

I.e. only 4 mount points instead of 37.

Indeed, the only remaining mount points held by the new NS are the /proc mounts (as requested by -b /proc), the rootfs '/' (necessary as it is the parent of /proc, and the actual device containing the new rootfs in /dev/shm/rootfs (tmpfs here).

**Comment by mickael-guene** [comment]
Fix in commit df0abe6f7b5edeaf5960724a99373c5b8592144e
